src/app/problems/problem-view/problem-view.component.html

```diff
@@ -14,6 +15,18 @@
         <td>{{uqj?.question?.token_value || 'No Token Value'}}</td>
         <td>{{uqj?.status || 'No Status'}}</td>
         <td>{{uqj?.question?.max_submission_allowed || 'No Value'}}</td>
+        <td>
+            <ng-container>
+                <button type="button" mat-icon-button
+                        [ngClass]="{'red-icon':favoriteStatus[uqj?.question.id], 'gray-icon':!favoriteStatus[uqj?.question.id]}"
```

Comment:
same thing as before. Make it just one variable since both of these classes are mutually exclusive